core/src/main/java/org/apache/accumulo/core/metadata/schema/ExternalCompactionFinalState.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.metadata.schema;
+
+import java.util.Base64;
+
+import org.apache.accumulo.core.data.TableId;
+import org.apache.accumulo.core.dataImpl.KeyExtent;
+import org.apache.accumulo.core.util.TextUtil;
+import org.apache.hadoop.io.Text;
+
+import com.google.common.base.Preconditions;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+public class ExternalCompactionFinalState {
+
+  private static final Gson GSON = new GsonBuilder().create();
+
+  public enum FinalState {
+    FINISHED, FAILED
+  }
+
+  private ExternalCompactionId ecid;
+  private KeyExtent extent;
+  private FinalState state;
+  private long fileSize;
+  private long fileEntries;
```

Comment:
These fields could be final
```suggestion
  private final ExternalCompactionId ecid;
  private final KeyExtent extent;
  private final FinalState state;
  private final long fileSize;
  private final long fileEntries;
```